lib/Myriad/RPC/Implementation/Redis.pm

```diff
@@ -84,16 +86,43 @@ async method stop () {
 async method create_group ($rpc) {
     unless ($rpc->{group}) {
         await $self->redis->create_group($rpc->{stream}, $self->group_name);
+        await $self->process_pending($rpc);
         $rpc->{group} = 1;
     }
 }
 
+async method process_pending ($rpc) {
+    my $pending_info = await $self->redis->pending_messages_info($rpc->{stream}, $self->group_name);
+    while ($pending_info->[0] > 0) {
+        my @items = await $self->redis->pending(stream => $rpc->{stream}, group => $self->group_name, client => $self->whoami);
+
+        for my $item (@items) {
+            push $item->{data}->@*, ('transport_id', $item->{id});
+            if ($rpc->{process_pending}){
+                try {
+                    $log->infof('processing pending request %s',$item);
+                    my $message = Myriad::RPC::Message::from_hash($item->{data}->@*);
+                    $rpc->{sink}->emit($message);
```

Comment:
I think we should start checking `$message->deadline` here (also in the regular delivery)